### PR TITLE
new "catch the egg" game

### PR DESCRIPTION
**Pull request by Shoom**
Bear with me, I didn't code in a looong time.
This is my first draft of a new game and just want to get some feedback. :P

Red "eggs" fall from the sky and you must position yourself (via bottom row left and right buttons) to catch it. Every catch speeds up the game.

### CODE REVIEW

**Comment by MartinHelmut** [comment]
Really cool! I definitely will play this game! 😎I only got one question: is it written for the new or old "game engine"?

**Comment by MartinHelmut** [comment]
Haha, just forget it! Definitely the new engine 😊

File: src/games/crimson.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/* eslint-disable */
```
Comment (arndissler) [inline src/games/crimson.js]: Can we remove the comment? If you remove the two semicolons in line 7 and line 51 the linter will not complain about anything.

File: src/games/crimson.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    }, wait);
+}
+
+function hit(){
+    playerHealth--;
+    if(playerHealth<0){
+        dead = true;
+        pause(1000,die);
+    } else{
+        resetEgg();
```
Comment (arndissler) [inline src/games/crimson.js]: Do we really need an `else` branch? The process will exit the game and will display the menu, so maybe add `return` and omit the `else`.

File: src/games/crimson.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+let leftPaddlePressed = false;
+let rightPaddlePressed = false;
+
+let eggX = 0;
+let eggY = 0;
+let eggAbsoluteY = 0;
+let eggColor = 'crimson';
+
+let speed = .025;
+let acceleration = .005;
```
Comment (arndissler) [inline src/games/crimson.js]: You can use `const` here

File: src/games/crimson.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+};
+
+let die;
+let dead = false;
+let paused = false;
+
+let playerPosition = 13;
+let playerColor = ['red','orange','olivedrab'];
+let playerHealth = playerColor.length-1;
+let leftPaddle = 12;
```
Comment (arndissler) [inline src/games/crimson.js]: You can use `const` here.

File: src/games/crimson.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+let rightPaddle = 15;
```
Comment (arndissler) [inline src/games/crimson.js]: minus `let`, plus `const`

File: src/games/crimson.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+function clamp(value, min, max) {
+    return Math.min(Math.max(value, min), max);
+};
+
+let die;
+let dead = false;
+let paused = false;
+
+let playerPosition = 13;
+let playerColor = ['red','orange','olivedrab'];
```
Comment (arndissler) [inline src/games/crimson.js]: You can use `const` for `playerColor`.

File: src/games/crimson.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+let playerHealth = playerColor.length-1;
```
Comment (arndissler) [inline src/games/crimson.js]: You should move the initialization of `playerHealth` and everything else for initializing the game to a startup routine and call it right after the player died and just before returning to the main menu (somewhere after line 69)

File: src/games/crimson.js (hunk 2 above)
Comment (Shoom) [reply src/games/crimson.js]: Thanks for catching that. The egg even needs to be resetted once your dead, so resetEgg() needs to be called in any case. ;)

Comment (arndissler) [reply src/games/crimson.js]: Yass. So just remove the `else` :)

File: src/games/crimson.js (hunk 7 above)
Comment (Shoom) [reply src/games/crimson.js]: And initially call it like this?
```
let initialized = false;

…

if(!initialized){
  init();
  initialized = true;
}
```

Comment (arndissler) [reply src/games/crimson.js]: Yes, but you have to deinitialize it when the playerHealth is below zero. Maybe prepare the board for the next round right after the playerHealth dropped?

```javascript
if(playerHealth<0){
  dead = true;
  pause(1000,die);
  // TODO: set all states to #000
  // TODO: set playerHealth to initial value
}
```

File: src/games/crimson.js [hunk 8]
```diff
@@ -66,17 +68,19 @@ function hit(){
     playerHealth--;
     if(playerHealth<0){
         dead = true;
-        pause(1000,die);
-    } else{
-        resetEgg();
+        pause(1000,function(){
+            die();
+            init();
```
Comment (arndissler) [inline src/games/crimson.js]: Just flip `initialized` would be enough :)

**Pull request merged**